Extension Array blogpost

TomAugspurger opened the pull request.

wesm commented on content/release-0.24.0.md:
```diff
@@ -1,3 +1,142 @@
+the "wrapper around a NumPy array" part. It'd be more correct to say "wrapper
+around an array-like object".
+
+While pandas mostly used NumPy's type system, we restricted it in places and
```
The term "type system" is one that seems to people seem to have different interpretations of. It might be more accurate to say "NumPy's built-in data representations" (where a dtype in NumPy has a 1-1 mapping onto a data representation)

wesm commented on content/release-0.24.0.md:
```diff
@@ -1,3 +1,142 @@
+is a way to take any array-like data structure (one or more NumPy arrays, an
+Apache Arrow array, a CuPy array) and place it inside a DataFrame. I think
+getting pandas out of the array business, and instead thinking about
+higher-level tabular data things, is a health development for the project.
```
Do you want to make a list of other known projects that are already using or testing out ExtensionArrays?

wesm commented on content/release-0.24.0.md:
```diff
@@ -1,3 +1,142 @@
+
+Extension arrays provide some interesting opportunities for pandas. 0.24.0
+introduced `IntegerArray`, a nullable-integer dtype, solving one of
+pandas-longest running feature requests.
```
This is a big enough deal that you might talk about it in a section in the main text; some people might feel like you're buring the lede =)

jorisvandenbossche commented on content/release-0.24.0.md:
```diff
@@ -1,3 +1,142 @@
 Date: 2019-01-04
 
+Extensibility has been of the major themes in pandas development over the last
+couple of releases. This post introduces the extension array interface: the
```
do we still want to start calling it the "pandas array interface" instead, or is that too late?

TomAugspurger replied:
I kept it as "extension array interface", since I use extension arrays (the concrete thing) in the next sentence.

jorisvandenbossche replied:
Maybe we can then try to call it "the pandas extension array interface" (including pandas), it's long, but I personally like it is a bit more explicit.

jorisvandenbossche commented on content/release-0.24.0.md:
```diff
@@ -1,3 +1,142 @@
+>>> ser.values
+[a, b, a]
+Categories (3, object): [a, b, c]
+>>> ser = pd.Series(
```
leftover here I think

jreback commented on content/release-0.24.0.md:
```diff
@@ -1,3 +1,142 @@
 Title: Pandas Extension Arrays
 Date: 2019-01-04
 
+Extensibility has been of the major themes in pandas development over the last
```
one of

jreback commented on content/release-0.24.0.md:
```diff
@@ -1,3 +1,142 @@
+Extensibility has been of the major themes in pandas development over the last
+couple of releases. This post introduces the extension array interface: the
+motivation behind them and how they might affect you as a pandas user. Finally,
+we look at how extension arrays shape the future of pandas.
```
will shape the future

jreback commented on content/release-0.24.0.md:
```diff
@@ -1,3 +1,142 @@
+Pandas is built on top of NumPy. You could roughly define a Series as a wrapper
+around a NumPy array, and a DataFrame as a collection of Series with a shared
+index. That's not entirely correct for several reasons, but I want to focus on
+the "wrapper around a NumPy array" part. It'd be more correct to say "wrapper
```
say It would (rather than a contraction)

jreback commented on content/release-0.24.0.md:
```diff
@@ -1,3 +1,142 @@
+addresses. I proposed an [IPArray][IPArray], but the community was hesitant to
+take on the maintainence burden of adding another custom dtype. So rather than
+accept an `IPArray` in pandas itself, we decided to implement an interface. Any
+object implementing this interface would be allowed in pandas. We'd let it into
```
contraction

jreback commented on content/release-0.24.0.md:
```diff
@@ -1,3 +1,142 @@
+implementation. We are *not* reimplementing NumPy here in pandas. Rather, this
+is a way to take any array-like data structure (one or more NumPy arrays, an
+Apache Arrow array, a CuPy array) and place it inside a DataFrame. I think
+getting pandas out of the array business, and instead thinking about
```
healthy

TomAugspurger commented:
Thanks for the feedback. Sorry about the rough quality earlier, it was a WIP.

I changed the intent of the post while writing. My main goal here is to explain the motivation behind extension arrays. I think the whatsnew is still the best place for "announcement". 

I've updated with a largish rewrite of the second half. The first part, up till Integer Array, is basically the same.

TomAugspurger commented:
Any objections to merging? I'll publicize this post along with the 0.24.0 release.

jreback commented:
lgtm

jorisvandenbossche commented:
Really nice post! I made a few minor corrections.

jorisvandenbossche commented on content/extension-arrays.md:
```diff
@@ -177,9 +176,9 @@ Apache Arrow array, a CuPy array) and place it inside a DataFrame. I think
-This dovetails perfectly with NumPy's [`__array_ufunc__`][ufunc] protocol and
-[NEP-18][nep18]. You'll be able to use the familiar NumPy API on objects that
-aren't backed by NumPy memory.
+This fits together perfectly with NumPy's [`__array_ufunc__`][ufunc] protocol
```
As a non-native speaker, I had never heard of "dovetails" before :-) 
I think the 'fits together perfectly' covers the same meaning?

TomAugspurger replied:
Thanks. I went with "works perfectly" Arther than "fits together perfectly"
